sdc-plugin/sdc_writer.cc

```diff
@@ -46,15 +46,19 @@ void SdcWriter::WriteSdc(RTLIL::Design* design, std::ostream& file) {
 
 void SdcWriter::WriteClocks(RTLIL::Design* design, std::ostream& file) {
     for (auto& clock : Clocks::GetClocks(design)) {
-	// FIXME: Input port nets are not found in VPR
 	auto& clock_wire = clock.second;
+	// FIXME: Input port nets are not found in VPR
 	if (clock_wire->port_input) {
 	    continue;
 	}
+	// Write out only GENERATED and EXPLICIT clocks
+	if (Clock::IsPropagated(clock_wire)) {
+	    continue;
+	}
 	file << "create_clock -period " << Clock::Period(clock_wire);
 	file << " -waveform {" << Clock::RisingEdge(clock_wire) << " "
 	     << Clock::FallingEdge(clock_wire) << "}";
-	file << " " << Clock::WireName(clock_wire);
```

Comment:
Can you explain why this is called `SourcePinName`, rather than `SourceWireName`?  Does Yosys have a first class cell pin object?

Reply:
`SOURCE_PIN` was the attribute in Vivado, but we do use Wires instead of pins. Hence, I changed the attribute and method name accordingly.

Reply:
Can yosys return the wire attached to a cell pin? If so, it might make more sense to stick with `SOURCE_PIN`, and change this code to get the wire from the source pin?